indexer.go

```diff
@@ -25,7 +32,117 @@ func NewIndexer(provider Provider, writer Writer) *Indexer {
 	}
 }
 
-// IndexBlock transfers all transactions to a known strcuture and saves them
+// Transaction struct handler of all transaction fields to be indexed
+type Transaction struct {
+	ID              int
+	Hash            string
+	FromAddress     string
+	ToAddress       string
+	AppPubKey       string
+	Blockchains     []string
+	MessageType     string
+	Height          int
+	Index           int
+	Proof           *provider.TransactionProof
+	StdTx           *provider.StdTx
+	TxResult        *provider.TxResult
+	Tx              string
+	Entropy         int
+	Fee             int
+	FeeDenomination string
+}
+
+func convertProvTransactionToTransaction(provTransaction *provider.Transaction) *Transaction {
```

Comment:
What is Prov? I can't really figure out what it means, so I think we should add some docs or pick a different function name.